File: src/main/java/com/github/MrRogerHuang/DefaultModel.java
package com.github.MrRogerHuang;

import java.io.Serializable;
import java.util.List;

public class DefaultModel implements Serializable {
    private final List<String> pluginClassNames;
    private final List<String> classpaths;
    
    public DefaultModel(List<String> pluginClassNames, List<String> classpaths) {
        this.pluginClassNames = pluginClassNames;
        this.classpaths = classpaths;
    }
    
    public boolean hasPlugin(Class type) {              
        return pluginClassNames.contains(type.getName());
    }
    
    public List<String> getClasspaths() { return classpaths; }
}

File: src/main/kotlin/com/github/MrRogerHuang/DefaultModel.kt
package com.github.MrRogerHuang

import java.io.Serializable

class DefaultModel(private val pluginClassNames: List<String>, val classpaths: List<String>) : Serializable {
    fun hasPlugin(type: Class<*>): Boolean {
        return pluginClassNames.contains(type.name)
    }
}